src/Php74/Php74.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Polyfill\Php74;
+
+/**
+ * @author Ion Bazan <ion.bazan@gmail.com>
+ *
+ * @internal
+ */
+final class Php74
+{
+    /**
+     * @param object|mixed $obj
+     *
+     * @return array|null
+     */
+    public static function get_mangled_object_vars($obj)
+    {
+        if (!\is_object($obj)) {
+            trigger_error('get_mangled_object_vars() expects parameter 1 to be object, '.\gettype($obj).' given', E_USER_WARNING);
+
+            return null;
+        }
+
+        $result = array();
+
+        if ($obj instanceof \ArrayIterator || $obj instanceof \ArrayObject) {
+            $reflector = new \ReflectionClass($obj instanceof \ArrayIterator ? 'ArrayIterator' : 'ArrayObject');
+            $flags = $reflector->getMethod('getFlags')->invoke($obj);
+            $reflector = $reflector->getMethod('setFlags');
+
+            $reflector->invoke($obj, ($flags & \ArrayObject::STD_PROP_LIST) ? 0 : \ArrayObject::STD_PROP_LIST);
+            $arr = (array) $obj;
+            $reflector->invoke($obj, $flags);
+        } else {
+            $arr = (array) $obj;
+        }
+
+        foreach ($arr as $key => $value) {
+            $result[(string) $key] = $value;
+        }
+
+        return $result;
```

Comment:
`return array_combine(array_keys($arr), array_values($arr));` would fix the issue with references, if we care. (do we?)

Reply:
Wow, thanks for this! :clap: